configure.ac

```diff
@@ -22,6 +22,12 @@ AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
 AC_CANONICAL_HOST
 
+# https://github.com/rsyslog/rsyslog/issues/1179
+# Seems libtool doesn't honors ARFLAGS, but statically set AR_FLAGS=cru
+# This causes a few warnings on build:
+#   "ar u modifier ignored since D is the default (see U)"
+m4_divert_text([DEFAULTS], [: "${ARFLAGS=Ucru} ${AR_FLAGS=Ucru}"])
```

Comment:
As I don't have centos/freebsd to try, I wasn't able to test this.
Anyway, setting:
> m4_divert_text([DEFAULTS], [: "${ARFLAGS=cr} ${AR_FLAGS=cr}"])

instead could work.
May I retry?